File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        pos = []

        # Store positions of all 1s
        for i, ch in enumerate(s):
            if ch == '1':
                pos.append(i)

        # Not enough 1s
        if len(pos) < k:
            return ""

        min_len = float('inf')
        ans = ""

        # Check every k consecutive 1s
        for i in range(len(pos) - k + 1):
            l = pos[i]
            r = pos[i + k - 1]

            cur = s[l:r + 1]
            length = r - l + 1

            if length < min_len:
                min_len = length
                ans = cur
            elif length == min_len and cur < ans:
                ans = cur

        return ans

File: 2904-shortest-and-lexicographically-smallest-beautiful-string/2904-shortest-and-lexicographically-smallest-beautiful-string.cpp
class Solution {
public:
    string shortestBeautifulSubstring(string s, int k) {
        vector<int> pos;

      
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '1')
                pos.push_back(i);
        }

      
        if (pos.size() < k)
            return "";

        int minLen = INT_MAX;
        string ans = "";

      
        for (int i = 0; i + k - 1 < pos.size(); i++) {
            int l = pos[i];
            int r = pos[i + k - 1];

            int len = r - l + 1;

            string cur = s.substr(l, len);

            if (len < minLen) {
                minLen = len;
                ans = cur;
            }
            else if (len == minLen && cur < ans) {
                ans = cur;
            }
        }

        return ans;
    }
};